test/replication-luatest/gh_6126_auto_rejoin_test.lua

```diff
@@ -0,0 +1,55 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local fio = require('fio')
+
+local g = t.group('gh_6126')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    local master_uri = server.build_instance_uri('master')
+    local replica_uri = server.build_instance_uri('replica')
+    local replication = {master_uri, replica_uri}
+    local box_cfg = {
+        replication = replication,
+    }
+    cg.replica = cg.cluster:build_and_add_server(
+        {alias = 'replica', box_cfg = box_cfg})
+
+    box_cfg.replication = {master_uri}
+    cg.master = cg.cluster:build_and_add_server(
+        {alias = 'master', box_cfg = box_cfg})
+
+    cg.master:start()
+end)
+
+g.after_each(function(cg)
+    cg.cluster.servers = nil
+    cg.cluster:drop()
```

Comment:
Let's set `cluster.servers` to nil after `cg.cluster:drop()`.
`cg.cluster:drop()` iterates over `cluster.servers` to clean them up.